### DISCUSSION TITLE
Only allow json requests to the API after 0.17.0

### DISCUSSION THREAD

**Discussion by matmair** [inventree/InvenTree#8661] [Ideas]
We currently accept `application/json`, `application/x-www-form-urlencoded` and `multipart/form-data` on our API endpoints. I want to remove all options but json to streamline corner cases with encoding and reduce the schema.
I have not seen a valid use case that can not use json instead of the other 2.

Thoughts @inventree/triage @inventree/maintainer ?

**Comment by SchrodingersGat** [comment]
Don't we need multipart for uploading files? We have custom code in the react forms framework to accommodate this

**Reply by matmair** [reply]
My idea would be to only allow form-data on endpoints that need data upload - those should be reduced to a few for attachments and imports, so the API surface should still be reduced a lot (over half at least).
